sdk/core/core-amqp/test/retry.spec.ts

```diff
@@ -17,347 +18,378 @@ const should = chai.should();
 import * as dotenv from "dotenv";
 dotenv.config();
 
-describe("retry function", function() {
-  it("should succeed if the operation succeeds.", async function() {
-    let counter = 0;
-    try {
-      const config: RetryConfig<any> = {
-        operation: async () => {
-          debug("counter: %d", ++counter);
-          await delay(200);
-          return {
-            code: 200,
-            description: "OK"
-          };
-        },
-        connectionId: "connection-1",
-        operationType: RetryOperationType.cbsAuth
-      };
-      const result = await retry(config);
-      result.code.should.equal(200);
-      result.description.should.equal("OK");
-      counter.should.equal(1);
-    } catch (err) {
-      debug("An error occurred in a test that should have succeeded: %O", err);
-      throw err;
-    }
-  });
-
-  it("should fail if the operation returns a non retryable error", async function() {
-    let counter = 0;
-    try {
-      const config: RetryConfig<any> = {
-        operation: async () => {
-          debug("counter: %d", ++counter);
-          await delay(200);
-          throw translate({
-            condition: "amqp:precondition-failed",
-            description: "I would like to fail, not retryable."
-          });
-        },
-        connectionId: "connection-1",
-        operationType: RetryOperationType.management
-      };
-      await retry(config);
-    } catch (err) {
-      should.exist(err);
-      should.equal(true, err instanceof MessagingError);
-      err.message.should.equal("I would like to fail, not retryable.");
-      counter.should.equal(1);
-    }
-  });
+[RetryPolicy.ExponentialRetryPolicy, RetryPolicy.LinearRetryPolicy].forEach(
+  () =>
+    function(retryPolicy) {
+      describe(`retry function for ${retryPolicy}`, function() {
+        it("should succeed if the operation succeeds.", async function() {
+          let counter = 0;
+          try {
+            const config: RetryConfig<any> = {
+              operation: async () => {
+                debug("counter: %d", ++counter);
+                await delay(200);
+                return {
+                  code: 200,
+                  description: "OK"
+                };
+              },
+              connectionId: "connection-1",
+              operationType: RetryOperationType.cbsAuth,
+              delayInSeconds: retryPolicy === RetryPolicy.ExponentialRetryPolicy ? 0 : 15,
```

Comment:
Why 0 in case of exponential? That will result in no exponential increase at all right?
I would imagine that we don't have to make any changes to the `delayInSeconds` property in any of the tests